ps3.py

```diff
@@ -0,0 +1,374 @@
+def deal_hand(n):
+    """
+    Returns a random hand containing n lowercase letters.
+    ceil(n/3) letters in the hand should be VOWELS (note,
+    ceil(n/3) means the smallest integer not less than n/3).
+
+    Hands are represented as dictionaries. The keys are
+    letters and the values are the number of times the
+    particular letter is repeated in that hand.
+
+    n: int >= 0
+    returns: dictionary (string -> int)
+    """
+
+    hand = {}
+    num_vowels = int(math.ceil(n / 3))
+    for i in range(num_vowels - 1):
+        x = random.choice(VOWELS)
+        hand[x] = hand.get(x, 0) + 1
+    hand['*'] = hand.get('*', 1)
+    for i in range(num_vowels, n):
+        x = random.choice(CONSONANTS)
+        hand[x] = hand.get(x, 0) + 1
+    return hand
+
+
+def update_hand(hand, word):
+    """
+    Does NOT assume that hand contains every letter in word at least as
+    many times as the letter appears in word. Letters in word that don't
+    appear in hand should be ignored. Letters that appear in word more times
+    than in hand should never result in a negative count; instead, set the
+    count in the returned hand to 0 (or remove the letter from the
+    dictionary, depending on how your code is structured).
+
+    Updates the hand: uses up the letters in the given word
+    and returns the new hand, without those letters in it.
+
+    Has no side effects: does not modify hand.
+
+    word: string
+    hand: dictionary (string -> int)
+    returns: dictionary (string -> int)
+    """
+    word = word.lower()
+    hand_modified = hand.copy()
+    for letters in word:
+        if letters in hand_modified.keys():
+            hand_modified[letters] -= 1
+    list_keys = list(hand_modified.keys())
+    for keys in list_keys:
+        if hand_modified[keys] <= 0:
+            hand_modified.pop(keys)
+    return hand_modified
+
+
+def is_valid_word(word, hand, word_list):
+    """
+    Returns True if word is in the word_list and is entirely
+    composed of letters in the hand. Otherwise, returns False.
+    Does not mutate hand or word_list.
+
+    word: string
+    hand: dictionary (string -> int)
+    word_list: list of lowercase strings
+    returns: boolean
+    """
+    word = word.lower()
+    for letters in word:
+        if letters not in hand.keys() or word.count(letters) > hand[letters]:
+            return False
+
+    if '*' in word:
+        list_split = word.split('*')
+        list_possible_words = [str(list_split[0] + i + list_split[1]) for i in VOWELS]
+        for words in list_possible_words:
+            if words in word_list:
+                return True
+        return False
+
+    if word not in word_list:
+        return False
+    return True
+
+
+def calculate_handlen(hand):
+    """
+    Returns the length (number of letters) in the current hand.
+
+    hand: dictionary (string-> int)
+    returns: integer
+    """
+    len_hand = 0
+    for keys in hand:
+        len_hand += hand[keys]
+    return len_hand
```

Comment:
You could try to use function sum() here

Reply:
+